src/GovUk.Education.ExploreEducationStatistics.Notifier.Tests/Functions/SubscriptionManagerTests.cs

```diff
@@ -0,0 +1,554 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Functions;
+using GovUk.Education.ExploreEducationStatistics.Notifier.Configuration;
+using GovUk.Education.ExploreEducationStatistics.Notifier.Functions;
+using GovUk.Education.ExploreEducationStatistics.Notifier.Requests;
+using GovUk.Education.ExploreEducationStatistics.Notifier.Services;
+using GovUk.Education.ExploreEducationStatistics.Notifier.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Notifier.Types;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using Xunit;
+using static GovUk.Education.ExploreEducationStatistics.Notifier.Tests.Utils.NotifierTestUtils;
+using static GovUk.Education.ExploreEducationStatistics.Common.TableStorageTableNames;
+
+namespace GovUk.Education.ExploreEducationStatistics.Notifier.Tests.Functions;
+
+public class SubscriptionManagerFunctionTests(NotifierFunctionsIntegrationTestFixture fixture)
+    : NotifierFunctionsIntegrationTest(fixture)
+{
+    private static readonly GovUkNotifyOptions.EmailTemplateOptions EmailTemplateOptions = new()
+    {
+        ReleaseAmendmentPublishedId = "release-amendment-published-id",
+        ReleaseAmendmentPublishedSupersededSubscribersId = "release-amendment-published-superseded-subscribers-id",
+        ReleasePublishedId = "release-published-id",
+        ReleasePublishedSupersededSubscribersId = "release-published-superseded-subscribers-id",
+        SubscriptionConfirmationId = "subscription-confirmation-id",
+        SubscriptionVerificationId = "subscription-verification-id"
+    };
```

Comment:
Could the function and mocked services which are built in each test be instantiated at a single point in the constructor instead? Would save a lot of repetition in the "Arrange" portion of each test.

Reply:
They absolutely could, and I'd much prefer them that way myself to be honest. I left them like this because it's what we're doing elsewhere, and when I checked the styleguide for this sort of thing I found this:
![image](https://github.com/dfe-analytical-services/explore-education-statistics/assets/63285990/dab0cf71-a54f-4453-817a-4930d6fb2874)